internal/controller/controllers/bmh_agent_controller.go

```diff
@@ -1106,6 +1106,10 @@ func (r *BMACReconciler) getChecksumAndURL(ctx context.Context, log logrus.Field
 	if err != nil {
 		return checksum, url, err
 	}
+	//MGMT-10570 check that the master list is not empty before referencing it
+	if len(machineList.Items) == 0 {
+		return checksum, url, errors.New("There are no machines with master label")
```

Comment:
It might. Do you suggest to add a return value to this function that indicates shouldStop?

Reply:
I asked in the ticket and didn't get a clear replay for the question what should happen in this case, what is expected to happen @flaper87?
But definitely not a hotloop

Reply:
Sorry, I had missed both pings :pray: 

I replied on the issue but will comment here for the public. 

This should never happen in a regular baremetal ZTP flow, unless something changed in the flow. Please, share more details about the environment. Hypershiftr? CAPI? Something else? 

Let'st try to underrstand why there were no master nodes in the spoke cluster first :smile:

Reply:
It did not happened with the normal flow but in a dev machine. We  suspect that there were no labels on the master nodes. So yes, this is a far fetched scenario but why not check for edge cases?

Reply:
This was, as mentioned. a far fetched edge case. I think this occurred when trying to reinstall a spoke cluster using dev-scripts, the problem is that this is not consistent or easy to reproduce.

I simply raised this because of the unchecked assumption that was spotted in the code. I actually have no idea how to get into this situation again. I know it's possible because I have been there by accident at least once.